python-package/lightgbm/basic.py

```diff
@@ -2617,7 +2730,7 @@ def _reverse_update_params(self) -> "Dataset":
     def set_field(
         self,
         field_name: str,
-        data: Optional[Union[List[List[float]], List[List[int]], List[float], List[int], np.ndarray, pd_Series, pd_DataFrame, pa_Table, pa_Array, pa_ChunkedArray]]
+        data: Optional[_LGBM_SetFieldType],
```

Comment:
I manually pulled this out into a new type, `LGBM_SetFieldType`, because `ruff` wanted to split this into a one-type-per-line list, which I thought made the function signature difficult to read.